test/HostActivationTests/GivenThatICareAboutSharedFxMultilevelLookup.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Xunit;
+using FluentAssertions;
+using Microsoft.DotNet.CoreSetup.Test;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.SharedFxMultilevelLookup
+{
+    public class GivenThatICareAboutSharedFxMultilevelLookup
+    {
+        private RepoDirectoriesProvider RepoDirectories;
+        private TestProjectFixture PreviouslyBuiltAndRestoredPortableTestProjectFixture;
+
+        private string _currentWorkingDir;
+        // private string _userDir;
+        private string _executableDir;
+        private string _cwdSharedFxBaseDir;
+        // private string _userSharedFxBaseDir;
+        private string _exeSharedFxBaseDir;
+        private string _builtSharedFxDir;
+        private string _sharedFxVersion;
+        private string _cwdSelectedMessage;
+        // private string _userSelectedMessage;
+        private string _exeSelectedMessage;
+
+        public GivenThatICareAboutSharedFxMultilevelLookup()
+        {
+            string realCurrentWorkingDirectory = Directory.GetCurrentDirectory();
+            string multilevelDir = Path.Combine(realCurrentWorkingDirectory, "multilevelSharedFxLookup");
+            string targetRID = Environment.GetEnvironmentVariable("TEST_TARGETRID");
+
+            string builtDotnet = Path.Combine(realCurrentWorkingDirectory, "..", "..",
+                "artifacts", targetRID, "intermediate", "sharedFrameworkPublish");
+
+            _currentWorkingDir = Path.Combine(multilevelDir, "cwd");
+            _executableDir = Path.Combine(multilevelDir, "exe");
+
+            _cwdSharedFxBaseDir = Path.Combine(_currentWorkingDir, "shared", "Microsoft.NETCore.App");
+            _exeSharedFxBaseDir = Path.Combine(_executableDir, "shared", "Microsoft.NETCore.App");
+
+            // Completely remove remaining files from other builds
+            if (Directory.Exists(multilevelDir))
+            {
+                Directory.Delete(multilevelDir, true);
+            }
+
+            // Only the shared folder is necessary in cwd
+            Directory.CreateDirectory(_cwdSharedFxBaseDir);
+
+            // It's necessary to copy everything to build the fixture from exe dir
+            CopyDirectory(builtDotnet, _executableDir);
```

Comment:
Also, does CopyDirectory perform a recursive copy or only the copy of the contents of the specified folder?